shared/observability/src/airflow_shared/observability/traces/otel_tracer.py

```diff
@@ -256,7 +260,7 @@ def _new_span(
         tracer = self.get_tracer(component=component)
 
         if start_time is None:
-            start_time = timezone.utcnow()
+            start_time = datetime.datetime.now(tz=pendulum.UTC)
```

Comment:
If `pendulum` is compatible, would it be better to use `pendulum` instead of `datetime` here?
```suggestion
            pendulum.now(tz=pendulum.UTC)
```

If yes, then we could also remove the top level import of `datetime`.

Reply:
It should be compatible but the issue is that I have to add a dependency to `pendulum` to the `pyproject.toml` file. There used to be one but I removed it.

Reply:
well - there is still pendulum.UTC used here so you have not got rid of pendulum yet. We are generally depending a lot on pendulum anyway now, so I guess we should leave it here as dependency - we had some plans to get rid of it, especially that in modern python versions (3.10) the datetime support and timezone support  is generally fixed, but it would be breaking change if we get rid of pendulum in core airflow and task-sdk especially. This will be a separate effort, so I suggest we leave pendulum here and in pyproject.toml

Reply:
Silly me. Fixed.